src/test/fakeSuite/extension.test.ts

```diff
@@ -1,286 +1,330 @@
+
+            configuration.setBuildBeforeLaunch(false);
+            const launchConfigurations: string[] = ["bin\\InterestingSmallMakefile\\ARC H3\\Debug\\main.exe(str3a,str3b,str3c)",
+                                                    "bin\\InterestingSmallMakefile\\arch1\\Debug\\main.exe(str3a,str3b,str3c)",
+                                                    "bin\\InterestingSmallMakefile\\arch2\\Debug\\main.exe()"];
+            for (const config of launchConfigurations) {
+                await configuration.setLaunchConfigurationByName(vscode.workspace.rootPath + ">" + config);
+                let status: string = await launch.getLauncher().validateLaunchConfiguration(make.Operations.debug);
+                let launchConfiguration: configuration.LaunchConfiguration | undefined;
+                if (status === launch.LaunchStatuses.success) {
+                    launchConfiguration = configuration.getCurrentLaunchConfiguration();
+                }
+
+                if (launchConfiguration) {
+                    launch.getLauncher().prepareDebugCurrentTarget(launchConfiguration);
+                    launch.getLauncher().prepareRunCurrentTarget();
+                }
+            }
+
+            // A bit more coverage, "RelSize" and "RelSpeed" are set up
+            // to exercise different combinations of pre-created build log and/or make tools.
+            // No configure is necessary to be run here, it is enough to look at what happens
+            // when changing a configuration.
+            configuration.setConfigurationByName("InterestingSmallMakefile_windows_configRelSize");
+            configuration.setConfigurationByName("InterestingSmallMakefile_windows_configRelSpeed");
+
+            // InterestingSmallMakefile_windows_configRelSpeed constructs a more interesting build command.
+            configuration.setTargetByName("Execute_Arch3");
+            make.prepareBuildTarget("Execute_Arch3");
+
+            // Compare the output log with the baseline
+            // TODO: incorporate relevant diff snippets into the test log.
+            // Until then, print into base and diff files for easier viewing
+            // when the test fails.
+            let parsedPath: path.ParsedPath = path.parse(extensionLogPath);
+            let baselineLogPath: string = path.join(parsedPath.dir, "../InterestingSmallMakefile_windows_baseline.out");
+            let extensionLogContent: string = util.readFile(extensionLogPath) || "";
+            let baselineLogContent: string = util.readFile(baselineLogPath) || "";
+            let extensionRootPath: string = path.resolve(__dirname, "../../../../");
+            baselineLogContent = baselineLogContent.replace(/{REPO:VSCODE-MAKEFILE-TOOLS}/mg, extensionRootPath);
+            // fs.writeFileSync(path.join(parsedPath.dir, "base.out"), baselineLogContent);
+            // fs.writeFileSync(path.join(parsedPath.dir, "diff.out"), extensionLogContent);

```

Comment:
These lines seem useful. Maybe they can be controlled to run via a command line switch

Reply:
I realized that the output of the tests are already highlighting some useful diffs. I had those lines written for when I used asserts and when my tests were not completely correctly wired up. Then I was not seeing anything and I used these files a lot. I noticed that I didn't use them since I moved to expect and I wired the tests (there was a missing "main();" in runTests.ts). I'm also thinking not to remove the commented code yet. But not activate it under a switch yet.